module/VuFind/src/VuFind/RateLimiter/RateLimiterManagerFactory.php

```diff
@@ -207,4 +211,22 @@ protected function createRedisCache(array $storageConfig): StorageInterface
 
         return new CredisStorage($redis, $options);
     }
+
+    /**
+     * Create a cache for Turnstile results.
+     *
+     * @param array $config Rate limiter configuration
+     *
+     * @return ?StorageInterface
+     */
+    protected function createTurnstileCache(array $config): \Laminas\Cache\Storage\StorageInterface
+    {
+        $turnstileConfig = unserialize(serialize($config));
+        $storageOptions = $turnstileConfig['Storage']['options'] ?? [];
+        $storageOptions['namespace'] = $storageOptions['turnstileNamespace'] ?? 'Turnstile';
+        $cacheManager = $this->getService(\VuFind\Cache\Manager::class);
+        $cache = $cacheManager->getCache('object', $storageOptions['namespace']);
+        $cache->getOptions()->setTtl($storageOptions['turnstileTtl'] ?? 60 * 60 * 24);
```

Comment:
So `createTurnstileCache` is for the cache that records whether you have already taken a Turnstile challenge, and what the result was.  Using \VuFind\CacheManager is a problem; I definitely don't want to use a file-based cache for the same performance reason that it's not used for the rest of the RateLimiter.  I could use the `createCache` method above to create (yet) another in-memory cache instance for this purpose, but at minimum I'd have to mess with it a little to pass in the appropriate TTL, and there might be a better approach.